### PR TITLE
A few parsing changes

### PR DESCRIPTION
**Pull request by whot**
Started out with using `parse` as the main module which makes it a lot more readable in terms of what we're parsing. Especially because it does hex conversion directly instead of requiring an extra line later.

The rest is just bits I found while looking for the above.

### CODE REVIEW

File: parse_hut.py [hunk 1]
```diff
@@ -35,16 +36,22 @@ def parse_usages(usage_list):
...
-            continue
-        if '-' in usage:
+        if _parse('{:x}-{:x}\t{}', line):
             print(line)
             continue
-        usages[int(usage, 16)] = name
+
+        r = _parse('{usage:x}\t{name}', line)
+        assert r is not None
+        if 'reserved' in r['name'].lower():
```
Comment (bentiss) [inline parse_hut.py]: This should be moved up as we don't really want to show the reserved ranges in the `if _parse('{:x}-{:x}\t{}', line):` statement above

File: hid-replay.py [hunk 2]
```diff
@@ -108,15 +108,17 @@ def inject_events(self, wait_max_seconds=2):
...
             pass
-        print('Hit enter (re)start replaying the events')
-        sys.stdin.readline()
-        replay.inject_events()
-    except KeyboardInterrupt:
-        pass
-    finally:
         replay.destroy()
+    finally:
+        pass
```
Comment (bentiss) [inline hid-replay.py]: Is this really reasonable? :)

**Review by bentiss — CHANGES_REQUESTED** [review]
OK, there 2 changes I'd like in before merging. Seems good otherwise

File: hid-replay.py [hunk 3]
```diff
@@ -32,36 +33,31 @@ def __init__(self, filename):
...
-                    dev.phys = phys
+                    r = parse('P: {phys}', l)
+                    if r is not None:
+                        dev.phys = r['phys']
                 elif l.startswith('R:'):
-                    length, rdesc = l.split(' ', 2)[1:]
-                    rdesc = [int(r, 16) for r in rdesc.split()]
-                    assert int(length) == len(rdesc)
-                    dev.rdesc = rdesc
+                    r = findall(' {:x}', l)
```
Comment (bentiss) [inline hid-replay.py]: The first item in the list is the length of the rdesc. It should be parsed as a plain decimal and should be stripped out of the report descriptor.

File: hid-replay.py (hunk 2 above)
Comment (whot) [reply hid-replay.py]: we have to contribute to global warming somehow. a few spare cpu cycles was all I could think of.

**Comment by whot** [comment]
PR updated

**Review by bentiss — APPROVED** [review]

**Comment by bentiss** [comment]
Sounds good now

**Pull request merged**